R-package/tests/testthat/test_Predictor.R

```diff
@@ -524,8 +518,52 @@ test_that("predictions for multiclass classification are returned as matrix", {
     expect_true(is.matrix(pred))
     expect_equal(nrow(pred), nrow(X))
     expect_equal(ncol(pred), 3L)
-    pred <- predict(model, X, rawscore = TRUE)
+    pred <- predict(model, X, type = "raw")
     expect_true(is.matrix(pred))
     expect_equal(nrow(pred), nrow(X))
     expect_equal(ncol(pred), 3L)
 })
+
+test_that("predict type='class' returns predicted class for classification objectives", {
+    data(agaricus.train, package = "lightgbm")
+    X <- as.matrix(agaricus.train$data)
+    y <- agaricus.train$label
+    dtrain <- lgb.Dataset(X, label = y, params = list(max_bins = 5L))
+    bst <- lgb.train(
+        data = dtrain
+        , obj = "binary"
+        , nrounds = 5L
+        , verbose = VERBOSITY
+    )
+    pred <- predict(bst, X, type = "class")
+    expect_true(all(pred %in% c(0L, 1L)))
+
+    data(iris)
+    X <- as.matrix(iris[, -5L])
+    y <- as.numeric(iris$Species) - 1.0
+    dtrain <- lgb.Dataset(X, label = y)
+    model <- lgb.train(
+      data = dtrain
+      , obj = "multiclass"
+      , nrounds = 5L
+      , verbose = VERBOSITY
+      , params = list(num_class = 3L)
+    )
+    pred <- predict(model, X, type = "class")
+    expect_true(all(pred %in% c(0L, 1L, 2L)))
+})
+
+test_that("predict type='class' returns values in the target's range for regression objectives", {
+    data(agaricus.train, package = "lightgbm")
+    X <- as.matrix(agaricus.train$data)
+    y <- agaricus.train$label
+    dtrain <- lgb.Dataset(X, label = y, params = list(max_bins = 5L))
+    bst <- lgb.train(
+        data = dtrain
+        , obj = "regression"
+        , nrounds = 5L
+        , verbose = VERBOSITY
+    )
+    pred <- predict(bst, X, type = "class")
+    expect_true(all(!(pred %in% c(0.0, 1.0))))
```

Comment:
```suggestion
    expect_true(!any(pred %in% c(0.0, 1.0)))
```

Since the last time a commit on this PR successfully passed CI, LightGBM has upgraded its `lint` CI job to `{lintr}` v3.0.0 and added several new linters (#5303).

Please accept this suggestion to resolve the one linter warning raised in https://github.com/microsoft/LightGBM/runs/6995392727?check_suite_focus=true.

```text
[[1]]
/home/runner/work/LightGBM/LightGBM/R-package/tests/testthat/test_Predictor.R:568:17: warning: [outer_negation] !any(x) is better than all(!x). The former applies negation only once after aggregation instead of many times for each element of x.
    expect_true(all(!(pred %in% c(0.0, 1.0))))
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```